### PR TITLE
Pr/fabric direct: Added support to build providers with FABRIC_DIRECT

### PR DESCRIPTION
**Pull request by shantonu**
As suggested in #1873, added the support to build providers with FABRIC_DIRECT. Just enabled for sockets provider, it can be done similarly for other providers as well.
- Created sockets/include directory for sockets provider header files
- Added empty fi_direct* files and map file for sockets provider
- Modified include/rdma header files to include fi_direct* files if FABRIC_DIRECT is enabled and added a define to disable static inlines.
- Enabled FABRIC_DIRECT for sockets provider in Travis-CI
- Added the definition in fi_direct* files for gni provider so that it uses it's fi_direct verison of static inlines  the existing static inlines in include/rdma header files. @hppritcha can you please verify if FABRIC_DIRECT works with gni provider with this change?

@sayantansur @shefty @jithinjosepkl

### CODE REVIEW

File: prov/sockets/include/rdma/fi_direct.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2014 Intel Corporation. All rights reserved.
```
Comment (sayantansur) [inline prov/sockets/include/rdma/fi_direct.h]: nitpick. since we are adding this file, can we do 2016 here?

File: prov/sockets/include/rdma/fi_direct.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#ifndef _FI_DIRECT_H_
+#define _FI_DIRECT_H_
```
Comment (sayantansur) [inline prov/sockets/include/rdma/fi_direct.h]: Why is this needed?

**Comment by hppritcha** [comment]
I'm checking.  Do not merge this in till one of the gni provider folks has a chance to test this.

**Comment by hppritcha** [comment]
Hmm... looks like commit be92ea1b busted support for enable-direct in the gni provider.

**Comment by hppritcha** [comment]
@shantonu I opened a PR against your branch to fix the configury problem.  If you could merge that commit in to your PR the gni crew will be happy with this PR.

File: prov/sockets/include/rdma/fi_direct.h (hunk 1 above)
Comment (shantonu) [inline prov/sockets/include/rdma/fi_direct.h]: sure!

**Comment by shantonu** [comment]
@hppritcha, sure! thanks!

File: prov/sockets/include/rdma/fi_direct.h (hunk 2 above)
Comment (shantonu) [inline prov/sockets/include/rdma/fi_direct.h]: Not needed. Forgot to remove.

**Comment by shantonu** [comment]
@hppritcha, please let me know which commit needs to be added to this PR.

**Comment by hppritcha** [comment]
Its this commit:
https://github.com/hppritcha/libfabric/commit/1ef0e0b6fec2884f43c47855e54665ed2cd41cb7
but all you need to do is go to 
https://github.com/shantonu/libfabric/pull/1
and press the merge button and you'll get the commit appearing here.

**Comment by hppritcha** [comment]
github is great!

**Comment by shantonu** [comment]
Thanks @hppritcha! I think the PR is ready now.

**Comment by hppritcha** [comment]
yep, once travis is happy its good to go

**Comment by shefty** [comment]
:+1:

**Comment by shefty** [comment]
This PR now has conflicts.  Can you resolve these and re-submit?

**Comment by shantonu** [comment]
Resolved the conflicts. Should be good now!

**Comment by shefty** [comment]
thanks
